From 96bbca715bd3caa634e56803d70950931dcad333 Mon Sep 17 00:00:00 2001
From: Iliyan Malchev <malchev@gmail.com>
Date: Sat, 18 Jun 2022 16:23:50 +0000
Subject: [PATCH] Fetch YOSYS_DATDIR from the environment

Signed-off-by: Iliyan Malchev <malchev@google.com>
---
 Makefile        | 3 ++-
 kernel/yosys.cc | 2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 9f7e1da35..2da7bd918 100644
--- a/Makefile
+++ b/Makefile
@@ -85,6 +85,7 @@ UNITESTPATH := tests/unit
 all: top-all
 
 YOSYS_SRC := $(dir $(firstword $(MAKEFILE_LIST)))
+YOSYS_DATDIR ?= $(DATDIR)
 VPATH := $(YOSYS_SRC)
 
 CXXSTD ?= c++11
@@ -642,7 +643,7 @@ OBJS += kernel/fstdata.o
 endif
 
 kernel/log.o: CXXFLAGS += -DYOSYS_SRC='"$(YOSYS_SRC)"'
-kernel/yosys.o: CXXFLAGS += -DYOSYS_DATDIR='"$(DATDIR)"' -DYOSYS_PROGRAM_PREFIX='"$(PROGRAM_PREFIX)"'
+kernel/yosys.o: CXXFLAGS += -DYOSYS_DATDIR='"$(YOSYS_DATDIR)"' -DYOSYS_PROGRAM_PREFIX='"$(PROGRAM_PREFIX)"'
 
 OBJS += libs/bigint/BigIntegerAlgorithms.o libs/bigint/BigInteger.o libs/bigint/BigIntegerUtils.o
 OBJS += libs/bigint/BigUnsigned.o libs/bigint/BigUnsignedInABase.o
diff --git a/kernel/yosys.cc b/kernel/yosys.cc
index 64d2b4def..284e39c58 100644
--- a/kernel/yosys.cc
+++ b/kernel/yosys.cc
@@ -903,7 +903,7 @@ void init_share_dirname()
 		return;
 	}
 #    ifdef YOSYS_DATDIR
-	proc_share_path = YOSYS_DATDIR "/";
+	proc_share_path = std::getenv("YOSYS_DATDIR");
 	if (check_file_exists(proc_share_path, true)) {
 		yosys_share_dirname = proc_share_path;
 		return;
-- 
2.37.0.rc0.104.g0611611a94-goog

