--- a/heir/lattigo.bzl
+++ b/heir/lattigo.bzl
@@ -4,7 +4,7 @@
 load("@rules_heir//heir:heir_opt.bzl", "heir_opt")
 load("@rules_heir//heir:heir_translate.bzl", "heir_translate")
 
-def _make_split_preprocessing_libs(utils_name, generated_heir_opt_name, heir_translate_flags, common_deps, tags, data):
+def _make_split_preprocessing_libs(utils_name, generated_heir_opt_name, heir_translate_flags, common_deps, tags, data, importpath = None):
     utils_go_filename = utils_name + ".go"
     heir_translate(
         name = utils_name + "_translate",
@@ -15,6 +15,7 @@
     go_library(
         name = utils_name,
         srcs = [":" + utils_go_filename],
+        importpath = importpath,
         deps = common_deps,
         tags = tags,
         data = data,
@@ -30,6 +31,7 @@
         data = [],
         tags = [],
         deps = [],
+        importpath = None,
         split_preprocessing = True,
         **kwargs):
     """A rule for generating Lattigo code from an MLIR file.
@@ -91,9 +93,10 @@
 
     if split_preprocessing == True:
         utils_name = go_package_name + "_utils"
-        _make_split_preprocessing_libs(utils_name, generated_heir_opt_name, heir_translate_flags, common_deps, tags, data)
+        utils_importpath = importpath + "_utils" if importpath else None
+        _make_split_preprocessing_libs(utils_name, generated_heir_opt_name, heir_translate_flags, common_deps, tags, data, importpath = utils_importpath)
         common_deps = common_deps + [utils_name]
-        utils_import_path = native.package_name() + "/" + utils_name
+        utils_import_path = utils_importpath if utils_importpath else (native.package_name() + "/" + utils_name)
         heir_translate_flags = heir_translate_flags + ["--extra-imports=" + utils_import_path, "--emit-lattigo-preprocessed"]
     else:
         heir_translate_flags = heir_translate_flags + ["--emit-lattigo"]
@@ -108,6 +111,7 @@
     )
     go_library(
         name = go_package_name,
+        importpath = importpath,
         srcs = extra_srcs + [":" + generated_go_filename],
         deps = common_deps,
         tags = tags,
